packages/pubsub/src/subscription.js

```diff
@@ -375,6 +378,15 @@ Subscription.formatName_ = function(projectId, name) {
 };
 
 /**
+ * Generate a random name to use for a name-less subscription.
+ *
+ * @private
+ */
+Subscription.generateName_ = function() {
+  return 'autogenerated-' + uuid.v4();
```

Comment:
This is the method which generates the subscription name. I originally wanted a better prefix, such as `google-cloud-pubsub-`, but starting a sub name with `goog` is reserved.